### PR TITLE
1014 patient search query

### PR DESCRIPTION
**Pull request by clemens-msupply**
Part of #1014 

- Add an `identifier` filter to search names / patient by (`code` OR `national_health_number` OR `program_patient_id`)
- Make `apply_simple_string_or_filter!` more low level. Previously it operated on two table columns. Now it just works like `apply_simple_string_filter!` only with `or_filter`, `apply_simple_string_or_filter` is than used in the repo to have ORs for three table columns for the identifier filter.
- ORs need to be applied before AND queries. A test and fix has been applied for the `item` table `code_or_name` filter.

Note, with the dynamic diesel `or_filter` and `filter` method you can't actually do arbitrary AND / OR filters (I previously thought you can). For example: (a OR b) AND (c OR d) can't be expressed; `query.filter(a).or_filter(b).filter(c).or_filter(d)` would result in:
`(a OR b) AND c OR d`

### CODE REVIEW

File: server/repository/src/diesel_macros.rs [hunk 1]
```diff
@@ -169,54 +154,46 @@ macro_rules! apply_simple_string_filter {
...
-                $query = $query.or_filter($dsl_field_2.eq(value));
-            }
 
-            if let Some(value) = string_filter.like {
-                // Use case insensitive like
-                $query = $query.filter($dsl_field_1.ilike(format!("%{}%", value.clone())));
-                $query = $query.or_filter($dsl_field_2.ilike(format!("%{}%", value)));
-            }
-        }
+/// Example expand, when called with:
```
Comment (andreievg) [inline server/repository/src/diesel_macros.rs:177]: A comment here would be great (has to be before ands)

**Review by andreievg — APPROVED** [review]
That's very cool thanks.

I think the comment in the issue about how `_or_filter` works would be good to add to `diesel_macros.rs`

Also, nice typing in create_filtered_query, those are alway quite confusing and trial error (at least for me), if you even need to do a join with filter, I went to some trouble trying to get this to work in a PR review, just out of interest: https://github.com/openmsupply/open-msupply/compare/986-example

**Review by mark-prins — APPROVED** [review]
👍  looks great

File: server/repository/src/diesel_macros.rs (hunk 1 above)
Comment (clemens-msupply) [reply server/repository/src/diesel_macros.rs:179]: done

thanks for the link, yes it took me a while to figure it out correctly...

**Pull request merged**